add option for location of prefs and temp files #2779

kpaharelau opened the pull request:
--portable
puts what normally goes in ~/.beaker instead into the folder where the application is mounted - {path_to_beaker_application}/.beaker .

kpaharelau committed:
add option for location of prefs and temp files #2779

--portable
puts what normally goes in ~/.beaker instead into the folder where the application is mounted - {path_to_beaker_application}/.beaker .

scottdraves commented on core/src/main/java/com/twosigma/beaker/core/module/config/DefaultBeakerConfig.java:
```diff
@@ -94,7 +96,18 @@ public DefaultBeakerConfig(BeakerConfigPref pref, GeneralUtils utils)
+    if (pref.getPortable()){
+      String path = Paths.get("").toAbsolutePath().toString();
+      boolean endsWithSlash = path.endsWith(File.separator);
+      path = path.substring(0,
```
what does this do?  is this removing the last directory?
if so then better to use subpath and getNameCount?
we should not do so much detailed string manipulation of filenames.

kpaharelau replied:
good point

scottdraves committed:
expand documentation for portable command-line option

kpaharelau committed:
add option for location of prefs and temp files #2779

what does this do? is this removing the last directory?
if so then better to use subpath and getNameCount?
we should not do so much detailed string manipulation of filenames.

FIXED

kpaharelau committed:
Merge branch 'cit/2779' of https://github.com/twosigma/beaker-notebook into cit/2779

kpaharelau committed:
add option for location of prefs and temp files #2779

simplify code

kpaharelau committed:
add option for location of prefs and temp files #2779

remove debug information
